modules/tinymce/src/themes/silver/test/ts/module/StickyHeaderStep.ts

```diff
@@ -159,6 +160,64 @@ const testStickyHeader = (toolbarMode: ToolbarMode, toolbarLocation: ToolbarLoca
       editor.options.unset('toolbar_sticky_offset');
     });
   });
+
+  context('Test inline editor editor with toolbar_mode: ' + toolbarMode, () => {
+    const element: SugarElement<HTMLElement> = SugarElement.fromHtml(`<p id="content">${Strings.repeat('some content ', 1000)}</p>`);
+    const setupElement = () => {
+
+      Insert.append(SugarBody.body(), element);
+
+      return {
+        element,
+        teardown: () => {
+          Remove.remove(element);
+        }
+      };
+    };
+    const hook = TinyHooks.bddSetupFromElement<Editor>({
+      plugins: 'fullscreen',
+      base_url: '/project/tinymce/js/tinymce',
+      toolbar: 'align | fontsize | fontfamily | blocks | styles | insertfile | forecolor | backcolor ',
+      menubar: false,
+      inline: true
+    }, setupElement);
+
+    PageScroll.bddSetup(hook.editor, 5000);
+
+    before(async () => {
+      // Need to wait for a fraction for some reason on safari,
+      // otherwise the initial scrolling doesn't work
+      await Waiter.pWait(100);
+    });
+
+    it('TINY-10581: Scroll to the end of the content, open and close the editor, and then scroll to the top of the content the editor should not shrink', async () => {
+      const editor = hook.editor();
+      const getRect = (element: SugarElement<any>): DOMRect =>
+        (element as SugarElement<HTMLElement>).dom.getBoundingClientRect();
+
+      element.dom.focus();
+      element.dom.scrollIntoView(false);
+
+      editor.focus();
+      await UiFinder.pWaitForVisible('Wait for the editor to show', SugarBody.body(), '.tox-editor-header');
+      const toolbar = await UiFinder.pWaitFor('toolbar should be visible', SugarBody.body(), '.tox-editor-header');
+
+      const initialWidth = getRect(toolbar).width;
+
+      element.dom.blur();
+      await UiFinder.pWaitForHidden('Wait for the editor to show', SugarBody.body(), '.tox-editor-header');
+
+      editor.focus();
+      await UiFinder.pWaitForVisible('Wait for the editor to show', SugarBody.body(), '.tox-editor-header');
+
+      window.scrollTo(0, 5000);
+      // this is needed to give the time to the toolbar to resize
+      await Waiter.pWait(0);
```

Comment:
Can we change this to a waiter that waits for the toolbar size to change. Like assert.atLeast(something)

Reply:
replaced with a waiter for scroll